pkg/credentialconfig/config.go

```diff
@@ -0,0 +1,330 @@
+}
+
+// ReadDockercfgFile attempts to read a legacy dockercfg file from the given paths.
+// if searchPaths is empty, the default paths are used.
+func ReadDockercfgFile(searchPaths []string) (cfg DockerConfig, err error) {
+	if len(searchPaths) == 0 {
+		searchPaths = DefaultDockercfgPaths()
+	}
+
+	for _, configPath := range searchPaths {
+		absDockerConfigFileLocation, err := filepath.Abs(filepath.Join(configPath, configFileName))
+		if err != nil {
+			klog.Errorf("while trying to canonicalize %s: %v", configPath, err)
+			continue
+		}
+		klog.V(4).Infof("looking for .dockercfg at %s", absDockerConfigFileLocation)
+		contents, err := ioutil.ReadFile(absDockerConfigFileLocation)
+		if os.IsNotExist(err) {
+			continue
+		}
+		if err != nil {
+			klog.Errorf("while trying to read %s: %v", absDockerConfigFileLocation, err)
+			continue
+		}
+		cfg, err := ReadDockerConfigFileFromBytes(contents)
+		if err != nil {
+			klog.Errorf("couldn't get the config from %q contents: %v", absDockerConfigFileLocation, err)
+			continue
+		}
+
+		klog.V(4).Infof("found .dockercfg at %s", absDockerConfigFileLocation)
+		return cfg, nil
+
+	}
+	return nil, fmt.Errorf("couldn't find valid .dockercfg after checking in %v", searchPaths)
+}
+
+// ReadDockerConfigJSONFile attempts to read a docker config.json file from the given paths.
+// if searchPaths is empty, the default paths are used.
+func ReadDockerConfigJSONFile(searchPaths []string) (cfg DockerConfig, err error) {
+	if len(searchPaths) == 0 {
+		searchPaths = DefaultDockerConfigJSONPaths()
+	}
+	for _, configPath := range searchPaths {
+		absDockerConfigFileLocation, err := filepath.Abs(filepath.Join(configPath, configJSONFileName))
+		if err != nil {
+			klog.Errorf("while trying to canonicalize %s: %v", configPath, err)
+			continue
+		}
+		klog.V(4).Infof("looking for %s at %s", configJSONFileName, absDockerConfigFileLocation)
+		cfg, err = ReadSpecificDockerConfigJSONFile(absDockerConfigFileLocation)
+		if err != nil {
+			if !os.IsNotExist(err) {
+				klog.Errorf("while trying to read %s: %v", absDockerConfigFileLocation, err)
+			}
+			continue
+		}
+		klog.V(4).Infof("found valid %s at %s", configJSONFileName, absDockerConfigFileLocation)
+		return cfg, nil
+	}
+	return nil, fmt.Errorf("couldn't find valid %s after checking in %v", configJSONFileName, searchPaths)
+
+}
+
+//ReadSpecificDockerConfigJSONFile attempts to read docker configJSON from a given file path.
+func ReadSpecificDockerConfigJSONFile(filePath string) (cfg DockerConfig, err error) {
+	var contents []byte
+
+	if contents, err = ioutil.ReadFile(filePath); err != nil {
+		return nil, err
+	}
+	return readDockerConfigJSONFileFromBytes(contents)
+}
+
+// ReadDockerConfigFile read a docker config file from default path
+func ReadDockerConfigFile() (cfg DockerConfig, err error) {
+	if cfg, err := ReadDockerConfigJSONFile(nil); err == nil {
+		return cfg, nil
+	}
+	klog.Warningf("Defaulting to old config file")
```

Comment:
You should log the error here (Warningf is fine). There are paths in ReadDockerConfigJSONFile which will return an error but not log that error.

Reply:
Gotcha, fixed